lib/watir_drops/page_object.rb

```diff
@@ -97,8 +115,27 @@ def fill_form(model)
     def inspect
       '#<%s url=%s title=%s>' % [self.class, url.inspect, title.inspect]
     end
+
     alias selector_string inspect
 
+
+    def on_page?
+      if self.respond_to?('page_url')
+        Watir::Wait.until { page_url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'') == (@browser.url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'')) }
```

Comment:
Well, technically you could have `page_url { 'google.com }`, so yeah, some kind of parsing out of the http is necessary. (This would be a good test case to add). :)